src/projectController.ts

```diff
@@ -212,12 +212,15 @@ export class ProjectController implements vscode.Disposable {
         for (const source of sourceDirectory) {
             projects.push(await CMakeProject.create(workspaceContext, source, isMultiProjectFolder));
         }
-        await ProjectController.checkBuildDirectories(projects);
+        await ProjectController.checkBuildDirectories(projects, sourceDirectory);
         return projects;
     }
 
     private static duplicateMessageShown = false;
-    private static async checkBuildDirectories(projects: CMakeProject[]) {
+    private static async checkBuildDirectories(projects: CMakeProject[], sourceDirectory: string[]) {
```

Comment:
The setting is called `sourceDirectory` and in the `config` class it should be called that, but in code it would be better to name local variables as `sourceDirectories` instead so that we don't forget it's an array.